### PR TITLE
Replaces "istype"s with is_helpers macros

### PR DESCRIPTION
**Pull request by nicbn**
Replaces a lot of "istype"s with is_helpers macros, also adds some new is_helpers macros.

### CODE REVIEW

File: code/__DEFINES/is_helpers.dm [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 #define islist(L) (istype(L,/list))
 
+#define ismob(L) (istype(L, /mob))
```
Comment (ChangelingRain) [inline code/__DEFINES/is_helpers.dm]: this already exists; it's a byond proc

File: code/game/objects/effects/effect_system/effects_foam.dm [hunk 2]
```diff
@@ -1,9 +1,5 @@
 // Foam
```
Comment (lordpidey) [inline code/game/objects/effects/effect_system/effects_foam.dm]: Did you mean to change this file?

File: code/game/objects/items/weapons/tanks/watertank.dm [hunk 3]
```diff
@@ -178,16 +178,15 @@
 //ATMOS FIRE FIGHTING BACKPACK
 
 #define EXTINGUISHER 0
-#define RESIN_LAUNCHER 1
```
Comment (lordpidey) [inline code/game/objects/items/weapons/tanks/watertank.dm]: Was this filechange intentional?

File: code/game/objects/items/weapons/holosign_creator.dm [hunk 4]
```diff
@@ -74,14 +74,6 @@
 	creation_time = 30
 	max_signs = 6
 
-/obj/item/weapon/holosign_creator/atmos
```
Comment (lordpidey) [inline code/game/objects/items/weapons/holosign_creator.dm]: This looks unintentional.

File: code/game/objects/structures/holosign.dm [hunk 5]
```diff
@@ -65,26 +65,6 @@
 /obj/structure/holosign/barrier/engineering
```
Comment (lordpidey) [inline code/game/objects/structures/holosign.dm]: Also appears unintentionally changed

File: code/modules/mining/equipment/survival_pod.dm [hunk 6]
```diff
@@ -208,7 +208,7 @@
 	..(mapload, TRUE)
 
 /obj/machinery/smartfridge/survival_pod/accept_check(obj/item/O)
-	if(istype(O, /obj/item))
+	if(isitem(O))
```
Comment (lordpidey) [inline code/modules/mining/equipment/survival_pod.dm]: return(isitem(O))

**Review by lordpidey — CHANGES_REQUESTED** [review]
Looks like you changed a few files you didn't mean to.

There's also one return statement that could be simplified.

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (nicbn) [inline code/__DEFINES/is_helpers.dm]: Is its behaviour identical to istype(L, /mob)?

File: code/game/objects/effects/effect_system/effects_foam.dm (hunk 2 above)
Comment (nicbn) [inline code/game/objects/effects/effect_system/effects_foam.dm]: Didn't noticed it. I'm fixing it.

File: code/__DEFINES/is_helpers.dm (hunk 1 above)
Comment (Jordie0608) [inline code/__DEFINES/is_helpers.dm]: Essentially yes http://www.byond.com/docs/ref/info.html#/proc/ismob

File: code/__DEFINES/is_helpers.dm [hunk 7]
```diff
@@ -52,6 +52,16 @@
 
 #define isalienadult(A) (istype(A, /mob/living/carbon/alien/humanoid))
 
+#define isalienhunter(A) (istype(A, /mob/living/carbon/alien/humanoid/hunter))
+
+#define isaliensentinel(A) (istype(A, /mob/living/carbon/alien/humanoid/sentinel))
+
+//#define isaliendrone(A) (istype(A, /mob/living/carbon/alien/humanoid/drone))
```
Comment (lzimann) [inline code/__DEFINES/is_helpers.dm]: Either use this or remove commented code.

File: code/game/objects/structures/crates_lockers/closets/secure/engineering.dm [hunk 8]
```diff
@@ -39,7 +39,6 @@
 	new /obj/item/clothing/gloves/color/yellow(src)
 	new /obj/item/clothing/gloves/color/yellow(src)
 	new /obj/item/weapon/inducer(src)
-	new /obj/item/weapon/inducer(src)
```
Comment (lzimann) [inline code/game/objects/structures/crates_lockers/closets/secure/engineering.dm]: Accidental remove here.

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 9]
```diff
@@ -575,7 +575,7 @@
 	if(!L.len) //no requirements
 		return 1
 
-	if(!istype(I, /obj/item/weapon/card/id) && istype(I, /obj/item))
+	if(!istype(I, /obj/item/weapon/card/id) && istype(I))
```
Comment (lzimann) [inline code/modules/mob/living/silicon/robot/robot.dm]: Forgot to replace the 2nd `istype` with `isitem`

File: code/game/objects/structures/crates_lockers/closets/secure/engineering.dm (hunk 8 above)
Comment (Expletive) [inline code/game/objects/structures/crates_lockers/closets/secure/engineering.dm]: >implying it was an accident

@nicbn has it out for me!

**Pull request merged**